### PR TITLE
Collect Device Aware Metadata 

### PR DESCRIPTION
**Pull request by jumaallan**
Story: https://app.shortcut.com/smileid/story/15105/add-fraud-signals-metadata-to-the-sdk-android

## Summary

Provides a fully extensible metadata provider that automatically updates metadata based on various device states using pluggable updaters.

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

File: lib/src/main/java/com/smileidentity/compose/biometric/OrchestratedBiometricKYCScreen.kt [hunk 1]
```diff
@@ -45,6 +48,7 @@ fun OrchestratedBiometricKYCScreen(
     showInstructions: Boolean = true,
     useStrictMode: Boolean = false,
     extraPartnerParams: ImmutableMap<String, String> = persistentMapOf(),
+    metadata: SnapshotStateList<Metadatum> = LocalMetadataProvider.current,
```
Comment (JNdhlovu) [inline lib/src/main/java/com/smileidentity/compose/biometric/OrchestratedBiometricKYCScreen.kt:51]: @jumaallan do we need this as observable since it's not something we need to trigger recomposition?

**Review by JNdhlovu — CHANGES_REQUESTED** [review]
@jumaallan approach seems great I think feels more like compose I have a few questions

1. Why do we use SnapshotStateList do we need to trigger recompositions for metadata as well?
2. I see we have strings on the metadata name, we once had the enum it feels like something potentially to be missed if one thing needs to change or we have to change in multiple places
3.

File: sample/src/main/java/com/smileidentity/sample/activity/MainActivity.kt [hunk 2]
```diff
@@ -1,19 +1,49 @@
...
         installSplashScreen()
         super.onCreate(savedInstanceState)
         WindowCompat.setDecorFitsSystemWindows(window, false)
+        checkAndRequestLocationPermissions()
         setContent {
             RootScreen()
         }
     }
+
+    private fun checkAndRequestLocationPermissions() {
```
[the remark this answers is not in the document]
Comment (robin-smileid) [reply sample/src/main/java/com/smileidentity/sample/activity/MainActivity.kt:27]: I'll add a comment to make it clear to partners looking at it that it's optional and here only used for demo purposes

**Review by robin-smileid — APPROVED** [review]
